### DISCUSSION TITLE
Upgrading the client on CentOS Stream 8

### DISCUSSION THREAD

**Discussion by rrodrigueznt** [abraunegg/onedrive#2091] [General]
Hi! I included in the title of the discussion the OS of my server, although I guess my particular case can be to some extent generalized.

I upgraded the client to `onedrive v2.4.20-14-g7b4f1a4` as per [Upgrading the client](https://github.com/abraunegg/onedrive/blob/master/docs/INSTALL.md#upgrading-the-client). As I installed `onedrive v2.4.20` from source, I removed it by doing `sudo make uninstall`.

If I'm not wrong, the process deleted all my OneDrive related services unit files from `/usr/lib/systemd/system/` and create a new `onedrive.service` and `onedrive@.service` ones.

Please, did it work as expected? Must I recreate all my unit files after each upgrade? Thanks!

**Comment by abraunegg** [comment]
@rrodrigueznt 
> If I'm not wrong, the process deleted all my OneDrive related services unit files from `/usr/lib/systemd/system/` and create a new `onedrive.service` and `onedrive@.service` ones.
> 
> Please, did it work as expected? 

Correct - the uninstall process removes all systemd files, together with all other application components. As per the Makefile:

![image](https://user-images.githubusercontent.com/4956234/184703293-d80cd38c-b5a0-4c18-839f-2b71823e33e2.png)

The reason the uninstall process does this, is so that when uninstalling, no component is left lying around. This is the correct practice.

Now - why on an upgrade is the *current* recommendation to uninstall before install ?

This recommendation goes back to the early days of the client development, where the 'onedrive' binary, depending on OS platform + distribution + long before the client was being packaged - folk would have the 'skilion' client installed in one location, this client installed in another location, but based on system search path rules, the 'skilion' client would then be used, rather than the updated one.

This installation path issue in part was fixed when the client started using 'autoconf' for building in early 2019 (https://github.com/abraunegg/onedrive/pull/287) - so, the installation path *should* be correct each and every time, thus, the installation over the top should be theoretically a better path forward without having to perform the 'uninstall + install' routine.

Will update the install document so this is clearer.

> Must I recreate all my unit files after each upgrade? Thanks!

In this instance, yes - you will need to create your custom systemd files. 

Moving forward, you *should* be able to just use the CentOS package and upgrade from that, or, upgrade the client bu building from source and just installing over the top.

The caveat here is - if you end up with 2 x 'onedrive' binaries due to other factors - you will need to remove the older client version manually.

**Reply by rrodrigueznt** [reply]
Thanks for the detailed answer! To recreate my custom `systemd` files is a low price to pay for keeping learning! I followed the recommendation in the section [Upgrading the client](https://github.com/abraunegg/onedrive/blob/master/docs/INSTALL.md#upgrading-the-client) in the second current paragraph.

**Reply by abraunegg** [reply]
@rrodrigueznt 
This advice has also been updated via https://github.com/abraunegg/onedrive/commit/d4595beabaa22d604ba1f788df130be1e37aa154

![image](https://user-images.githubusercontent.com/4956234/184715255-d0491ec5-2ba6-4fe3-aa18-0e63307fc01c.png)

**Reply by rrodrigueznt** [reply]
Cool! Thanks. Please, would it be advisable to include a reference to deleting all custom `systemd` files if you follow the uninstallation/reinstallation pathway? Would it be possible to offer to keep those files at uninstallation time as an option? I guess the switch must also check if the unit files of the newer release must introduce any new parameter or syntax.

Do you think that option deserves a new feature request?

**Reply by abraunegg** [reply]
An uninstall *should* uninstall all items - custom or otherwise. It is not wise to leave these 'hanging' about.

I would not consider this for a feature request.

**Reply by rrodrigueznt** [reply]
I get your point. Perhaps a warning about the deletion? Thanks!
